### PR TITLE
converting to launchmon to autotools package

### PR DESCRIPTION
**Pull request by lee218llnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/launchmon/package.py [hunk 1]
```diff
@@ -4,30 +4,29 @@
...
-        configure(
-            "--prefix=" + prefix,
-            "--with-bootfabric=cobo",
-            "--with-rm=slurm")
-
-        make()
-        make("install")
+    @run_before('autoreconf')
+    def boostrap(self):
+        os.environ['ACLOCAL_PATH'] += ':' + os.path.join(self.spec['libgcrypt'].prefix, "share", "aclocal")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/launchmon/package.py]: Could we set this in `setup_environment` instead?

File: var/spack/repos/builtin/packages/launchmon/package.py [hunk 2]
```diff
@@ -4,30 +4,30 @@
...
 
-    def install(self, spec, prefix):
-        configure(
-            "--prefix=" + prefix,
-            "--with-bootfabric=cobo",
-            "--with-rm=slurm")
-
-        make()
-        make("install")
+    def setup_environment(self, spack_env, run_env):
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/launchmon/package.py]: Why is this method needed?

File: var/spack/repos/builtin/packages/launchmon/package.py [hunk 3]
```diff
@@ -4,30 +4,30 @@
...
+        #os.environ['ACLOCAL_PATH'] += ':' + os.path.join(self.spec['libgcrypt'].prefix, "share", "aclocal")
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/launchmon/package.py]: Please remove the commented-out line

File: var/spack/repos/builtin/packages/launchmon/package.py [hunk 4]
```diff
@@ -4,30 +4,30 @@
...
+        spack_env.prepend_path('ACLOCAL_PATH', os.path.join(
+            self.spec['libgcrypt'].prefix, "share", "aclocal"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/launchmon/package.py]: You can do `self.spec['libgcrypt'].prefix.share.aclocal`, no `os.path.join` necessary.

**Comment by lee218llnl** [comment]
@adamjstewart thanks for the suggestions, this helps a lot! Hopefully my latest commit resolves all of your requests

**Comment by lee218llnl** [comment]
@citibeth the bootstrapping fails to find the appropriate libgcrypt during the invocation of automake if ACLOCAL_PATH isn't set appropriately. Does that answer your question?

File: var/spack/repos/builtin/packages/launchmon/package.py [hunk 5]
```diff
@@ -4,30 +4,29 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
+import os
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/launchmon/package.py]: This is no longer needed.

**Comment by citibeth** [comment]
> @citibeth the bootstrapping fails to find the appropriate libgcrypt during the invocation of automake if ACLOCAL_PATH isn't set appropriately. Does that answer your question?

Unfortunately not... because the previous `MakefilePackage` didn't seem to do any setting of `ACLOCAL_PATH`, so I'm wondering why it's needed now?

**Comment by lee218llnl** [comment]
@citibeth This is required for the master branch which doesn't have the configure script generated. With the release version the configure script is already present and thus the boostrapping (automake) does not need to be invoked. Would you prefer it if I only prepend to the ACLOCAL_PATH when building the master branch?

**Comment by citibeth** [comment]
I think I prefer it the way it is, see #11706 and #11684.

I'd still like to see a comment on the `ACLOCAL_PATH` line.  Why does it need to reach inside of `libgcrypt` when running autotools?  Most Autotools packages don't need this kind of thing.

**Comment by lee218llnl** [comment]
@citibeth OK, I can do that. Let me do some testing and then I'll push another commit. Sorry for all the iterations!

**Comment by citibeth** [comment]
@lee218llnl I said I like it the way it is, without additional changes.  I just wanted to make sure everyone's happy with it before hitting "Merge".

**Comment by citibeth** [comment]
(To be more specific: it looks like the ONLY change I'm asking for is a comment; unless something doesn't work, of course).

**Comment by lee218llnl** [comment]
OK, I think this is set to go now. Thanks for putting up with all the edits!

**Pull request merged**